File: framework/src/gmsec5/field/BooleanField.cpp
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


#include <gmsec5/field/BooleanField.h>
#include <gmsec5/internal/field/InternalBooleanField.h>


using namespace gmsec::api5;
using namespace gmsec::api5::internal;


BooleanField::BooleanField(const char* name, bool value, bool isHeader)
	: m_internal(new InternalBooleanField(name, value, isHeader))
{
	registerChild(m_internal);
}


BooleanField::BooleanField(const BooleanField& other)
	: m_internal(new InternalBooleanField(other.getName(), other.getValue(), other.isHeader()))
{
	registerChild(m_internal);
}



BooleanField::~BooleanField()
{
	delete m_internal;
}


bool BooleanField::getValue() const
{
	return m_internal->getValue();
}


const char* BooleanField::toXML() const
{
	return m_internal->toXML();
}


const char* BooleanField::toJSON() const
{
	return m_internal->toJSON();
}

File: java/gov/nasa/gsfc/gmsec/api5/field/BooleanField.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file BooleanField.java
 */

package gov.nasa.gsfc.gmsec.api5.field;

import gov.nasa.gsfc.gmsec.api5.GmsecException;
import gov.nasa.gsfc.gmsec.api5.jni.field.JNIBooleanField;


/**
 * Specialized Field class that can be used to retain a boolean value.
 */
public class BooleanField extends Field
{
	//! @cond
	/**
	 * @hidden
	 * This constructor is for internal GMSEC API use only.
	 * @param field A JNIBooleanField object.
	 */
	public BooleanField(JNIBooleanField field)
	{
		setInternal(field);
	}
	//! @endcond


	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data Boolean data to associate with the field.
	 *
	 * @throws IllegalArgumentException Thrown if the field name string is null, is an empty string, or is not compliant.
	 */
	public BooleanField(String name, boolean data) throws IllegalArgumentException
	{
		this(name, data, false);
	}


	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data Boolean data to associate with the field.
	 * @param isHeader used to indicate if Field is a header field.
	 *
	 * @throws IllegalArgumentException Thrown if the field name string is null, is an empty string, or is not compliant.
	 */
	public BooleanField(String name, boolean data, boolean isHeader) throws IllegalArgumentException
	{
		validateName(name);

		setInternal(new JNIBooleanField(name, data, isHeader));
	}


	/**
	 * Copy constructor.
	 *
	 * @param other The other BooleanField object to copy.
	 *
	 * @throws IllegalArgumentException Thrown if the given BooleanField is null.
	 */
	public BooleanField(BooleanField other) throws IllegalArgumentException
	{
		if (other == null)
		{
			throw new IllegalArgumentException("BooleanField object is null");
		}

		setInternal(new JNIBooleanField((JNIBooleanField) Field.getInternal(other)));
	}


	/**
	 * Returns the data stored within the BooleanField.
	 *
	 * @return A boolean value.
	 */
	public boolean getValue()
	{
		JNIBooleanField jboolField = (JNIBooleanField) getInternal();

		return jboolField.getValue();
	}
}
